### PR TITLE
Support of config block

### PR DESCRIPTION
**Pull request by philsch**
Hi,

I've created an AngularJS provider around of your service to make feature flags configurable in the application's config phase. An example is added in the README. Configuration via .run() is still possible and old behavior shouldn't have changed at all.

Looking forward to your feedback :)

### CODE REVIEW

**Comment by mjt01** [comment]
Thanks for your contribution. Appreciate the effort you've gone to. There are a couple of things that would be good to tidy up.

- Coverage has dropped because you aren't exercising the provider in a unit test
- Rather than alter the eslint config i would prefer to define the function first.
- If the provider and service are to be in the same file, maybe rename the file to `.provider` rather than `.service` to be clear.
- Inside the provider we should be able to manage the `initialFlags` array as a local variable (see [Provider Recipe example here](https://docs.angularjs.org/guide/providers)
- As an alternative to the `init` method on the service could we initialise the `flags` variable inside the service as `flags = [] || initialFlags`?
- Prefer to bump version in a separate commit

**Comment by philsch** [comment]
Thanks for your suggestions :+1: 

- I would keep the `init` method because I can call the internal `set` method from there.. So any logic for setting flags are applied (maybe some more added in the future)
- the version is back to `v1.0.0` but I haven't adjusted the README.. please update the version mentioned there yourself

**Comment by philsch** [comment]
Hi @mjt01 

I know you've surely a lot of other stuff to do, but it would be really cool if you could check and merge my pull requests. 

Thank you very much!

**Comment by mjt01** [comment]
Sorry mate, wrapping up the biggest project of my career at the moment and
launching this week so I just haven't had the time to dedicate to it.

Will do my best to get onto it soon.
Thanks

File: .eslintrc.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ module.exports = {
     "no-undef": 2,
     "no-undefined": 2,
     "no-unused-vars": 2,
-    "no-use-before-define": 2,
+    "no-use-before-define": [2],
```
Comment (mjt01) [inline .eslintrc.js]: This change can be reverted as its not required with your final revision of this pull request (no longer triggering use before define error)

File: README.md [hunk 2]
```diff
@@ -60,6 +60,18 @@ myApp.run(function(featureFlags, $http) {
 });
 ```
 
+### Setting flag data on config phase (≥ v1.0.1)
```
Comment (mjt01) [inline README.md]: I was thinking this pull request is a bit bigger than a patch and was going to update it to v1.1.0

Can you update these v1.0.1 references to v1.1.0 please

**Comment by mjt01** [comment]
Thanks for being patient it's been a crazy couple of months. Apart from the minor tweaks could you squash this into one commit on the pr to keep the master tree clean please. Once you tidy these little things up i promise to cut the merge and cut the release asap.

Thanks again

**Comment by philsch** [comment]
Hi

Hope your work have been worth ;)

First time I squashed a git history, thanks for the learning. Hope I haven't missed something, but looks good so far. Your latest suggestions are included.

**Pull request merged**